lib/matplotlib/legend.py

```diff
@@ -535,6 +536,24 @@ def val_or_rc(val, rc_name):
                     loc = locs[0] + ' ' + locs[1]
             # check that loc is in acceptable strings
             loc = _api.check_getitem(self.codes, loc=loc)
+        elif isinstance(loc, tuple):
+            # validate the tuple represents Real coordinates
+            if len(loc) != 2 or not (all(isinstance(e, numbers.Real)
+                                         for e in loc)):
+                raise ValueError(
+                    "loc must be string, coordinate tuple or," +
+                    f" an integer 0-10, not {loc!r}")
```

Comment:
Since you have the same message three times, I suggest defining it as a variable before the if-loop.  Then you can just re-use it.

Reply:
Thanks, I have updated the PR with this change.